code/modules/paperwork/paper.dm

```diff
@@ -307,15 +309,11 @@
 			else
 				return
 
-		var/obj/item/pen/P = I
-		if(!P.active)
-			P.toggle()
-
-		if(P.iscrayon)
-			iscrayon = TRUE
-
-		if(P.isfancy)
-			isfancy = TRUE
+		if(!(I.get_tool_property(TOOL_PEN, TOOL_PROP_PEN_FLAG) & PEN_FLAG_ACTIVE))
+			var/decl/tool_archetype/pen/parch = GET_DECL(TOOL_PEN)
+			parch.toggle_active(usr, I)
+		iscrayon = I.get_tool_property(TOOL_PEN, TOOL_PROP_PEN_FLAG) & PEN_FLAG_CRAYON
```

Comment:
As above, you should keep the value of get_tool_property when checking it repeatedly instead of getting it repeatedly.